src/index.ts

```diff
@@ -694,6 +694,16 @@ export async function precomputedInit(
     skipInitialRequest = false,
   } = config;
 
+  // Add assignment cache initialization
+  const storageKeySuffix = buildStorageKeySuffix(apiKey);
+  const assignmentCache = assignmentCacheFactory({
+    chromeStorage: chromeStorageIfAvailable(),
+    storageKeySuffix,
+  });
+  if (assignmentCache instanceof HybridAssignmentCache) {
+    await assignmentCache.init();
+  }
```

Comment:
Uses a hybrid assignment cache for the fetch-based `precomputedInit` method

Reply:
why do we need to check on the type of `assignmentCache` here? I'd prefer if we didn't make any assumptions on the actual implementation. Can that `init()` call be handled inside the factory instead?

Reply:
This is copying a pattern that is already in place for the non-precomputed init methods, so I don't know the original intent. One benefit I see from looking at the code though is that it's here to keep the `await` out of the `assignmentCacheFactory`, which ensures that the factory method is synchronous and therefore can be used in both the online and offline init methods

Reply:
yeah this seems like necessary evil around mixing async and sync code.

I suppose another thing you could do is check to see if the cache has an init() method, and if so await init it. A little more future proof perhaps.